src/lib.rs

```diff
@@ -1,551 +1,493 @@
-    /// random value from app, 16 bytes
-    #[serde(with = "serde_bytes")]
-    nonce: Vec<u8>,
+pub struct SatsCard<T: Transport + Sized> {
+    pub transport: T,
+    pub proto: usize,
+    pub ver: String,
+    pub birth: usize,
+    pub slots: Vec<usize>,
+    pub addr: String,
+    pub pubkey: PublicKey,
+    pub card_nonce: Vec<u8>, // 16 bytes
 }
 
-impl CheckCommand {
-    pub fn new(nonce: Vec<u8>) -> Self {
-        CheckCommand {
-            cmd: "check".to_string(),
-            nonce,
+impl<T: Transport + Sized> SatsCard<T> {
+    pub fn new_slot(
+        &mut self,
+        slot: usize,
+        chain_code: Vec<u8>,
+        cvc: String,
+    ) -> Result<NewResponse, Error> {
+        let chain_code = Some(chain_code);
+        let pubkey = &self.pubkey;
+        let secp = self.transport.secp();
+        let nonce = self.card_nonce.to_vec();
+        let command = "new".to_string();
+        let ekeys_xcvc = calc_ekeys_xcvc(secp, pubkey, &nonce, cvc, command);
+        let epubkey = ekeys_xcvc.1.serialize().to_vec();
+        let xcvc = ekeys_xcvc.2;
+        let response = new_command(&self.transport, slot, chain_code, epubkey, xcvc);
+        if let Ok(new_response) = &response {
+            self.card_nonce = new_response.card_nonce.clone();
         }
+        response
     }
-}
 
-impl CommandApdu for CheckCommand {}
-
-/// New Command
-///
-/// SATSCARD: Use this command to pick a new private key and start a fresh slot. The operation cannot be performed if the current slot is sealed.
-///
-/// TAPSIGNER: This command is only used once.
-///
-/// The slot number is included in the request to prevent command replay.
-///
-/// At this point:
-///
-///     No new slots available? Abort and fail command.
-///     A new key pair is picked and stored into the new slot.
-///         The chain_code must be used in that process and stored.
-///         The card uses TRNG to pick a new master_pubkey (pair).
-///
-/// The new values take effect immediately, so some fields of the next status response will have
-/// new values.
-#[derive(Serialize, Clone, Debug, PartialEq, Eq)]
-pub struct NewCommand {
-    /// 'new' command
-    cmd: String,
-    /// (optional: default zero) slot to be affected, must equal currently-active slot number
-    slot: usize,
-    /// app's entropy share to be applied to new slot (optional on SATSCARD), 32 bytes
-    #[serde(with = "serde_bytes")]
-    chain_code: Option<Vec<u8>>,
-    /// app's ephemeral public key, 33 bytes
-    #[serde(with = "serde_bytes")]
-    epubkey: Vec<u8>,
-    /// encrypted CVC value, 6 bytes
-    #[serde(with = "serde_bytes")]
-    xcvc: Vec<u8>,
-}
-
-impl NewCommand {
-    pub fn new(slot: usize, chain_code: Option<Vec<u8>>, epubkey: Vec<u8>, xcvc: Vec<u8>) -> Self {
-        NewCommand {
-            cmd: "new".to_string(),
-            slot,
-            chain_code,
-            epubkey,
-            xcvc,
+    pub fn read(&mut self) -> Result<ReadResponse, Error> {
```

Comment:
Thoughts on using different constructors to build the request objects. Example here:

pub fn read(&mut self) -> Result<ReadResponse, Error> {
        let read_cmd = ReadCommand::for_satscard(self.card_nonce.clone());
        let read_response = self.transmit_read(read_cmd);
        if let Ok(read_response) = &response {
            self.card_nonce = read_response.card_nonce.clone();
        }
        response
    }